src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -196,6 +199,12 @@ export const resolveExternalBookReference = async(
 ) => {
   const bookInformation = await getBookInformation(services, reference);
 
+  // Don't throw an error if reference couldn't be loaded when UNLIMITED_CONTENT is truthy
+  // It will be processed in contentLinkHandler.ts
+  if (UNLIMITED_CONTENT && !bookInformation) {
+    return bookInformation;
+  }
```

Comment:
yes, if `UNLIMITED_CONTENT` is set to `true` we will process `undefined` data from `getBookInformation`
if `UNLIMITED_CONTENT` is not set then we will throw an error

Reply:
yea the code would be cleaner if `resolveExternalBookReference` returned a reference instead of a book, then you'd need less conditionals in `loadContentReference`, but the way this is set up is fine, its because of the way the logic was being done already